### ISSUE TITLE
No gradients provided for any variable when doing binarization

### ISSUE THREAD

**Issue by maxbry** [tensorflow/tensorflow#44125]
I have written a Lambda layer that converts an input variable of range `(0, 1)` to either 0 or 1, i.e. the layer binarizes the inputs. I did this by using `K.random_uniform(shape=K.shape(x)) <= x`. However, when wantint to train the model, I'm getting the error message, that no gradients are provided for any variable. How do I have to change the code such that my idea works? TF version is 2.2.0.

    import numpy as np

    from keras import backend as K
    from keras.layers import *
    from keras.models import Model

    def binarize(d):
        x1, x2 = d
        return x1 * K.cast(K.random_uniform(shape=K.shape(x2)) <= x2, 'float32')

    inp = Input((1,))
    var = Dense(1, activation='sigmoid')(inp)
    out = Lambda(binarize)([inp, var])

    model = Model(inp, out)
    model.compile(loss='mse', optimizer='sgd')

    x = np.random.normal(size=(128, 1))
    y = x >= 0
    model.train_on_batch(x, y)

**Comment by Saduf2019** [comment]
@maxbry 
I ran the code shared and face a different error, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/676f7c98d228a2a96d938831d6fd9d70/untitled446.ipynb), please share a colab gist if possible with the issue reported.

**Comment by maxbry** [comment]
[Here](https://colab.research.google.com/drive/1WnDD0CnhBdFLqolEnMuvfCjGZhLhCEt0) is a gist where the reported error is raised.

**Comment by Saduf2019** [comment]
@maxbry 
Could you please provide access to the colab.

**Comment by maxbry** [comment]
[Here](https://colab.research.google.com/drive/1WnDD0CnhBdFLqolEnMuvfCjGZhLhCEt0?usp=sharing)

**Comment by Saduf2019** [comment]
I am able to replicate this issue please find the [gist here](https://colab.research.google.com/gist/Saduf2019/34bd82ee685cacbc789d5ff9ec795de8/untitled450.ipynb).

**Comment by ymodak** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.
